### PR TITLE
aodn/backlog#2811

### PR DESCRIPTION
**Pull request by jonescc**
Make time and vis values optional

### CODE REVIEW

File: api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/OptionalIntegerFormatValidation.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    public OptionalIntegerFormatValidation(Function<StagedRow, String> getField, String columnTarget) {
+        super(getField, columnTarget, Collections.emptyList());
+        this.getField = getField;
+    }
+
+    @Override
+    public Validated<StagedRowError, Integer> valid(StagedRow target) {
+        if (StringUtils.isBlank(getField.apply(target))) {
+            return Validated.valid(null);
```
Comment (utas-tajona) [inline api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/OptionalIntegerFormatValidation.java:23]: what aout reutrning a Option<Integer>?

File: api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java [hunk 2]
```diff
@@ -16,6 +17,10 @@ public TimeFormatValidation() {
 
     @Override
     public Validated<StagedRowError, LocalTime> valid(StagedRow target) {
+        if (StringUtils.isBlank(target.getTime())) {
+            return Validated.valid(null);
+        }
```
Comment (utas-tajona) [inline api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java:22]: same here Optional<LocalTime>

**Review by utas-tajona — CHANGES_REQUESTED** [review]

File: api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java (hunk 2 above)
Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java:22]: Sure

File: api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/OptionalIntegerFormatValidation.java (hunk 1 above)
Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/OptionalIntegerFormatValidation.java:23]: Yes that would be better

Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/OptionalIntegerFormatValidation.java]: Fixed.

File: api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java (hunk 2 above)
Comment (jonescc) [reply api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/format/TimeFormatValidation.java:24]: Fixed

**Review by utas-tajona — APPROVED** [review]
LGTM

**Pull request merged**